test/SlowTests/Issues/RavenDB-21490.cs

```diff
@@ -0,0 +1,169 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using FastTests;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Exceptions.Corax;
+using Raven.Client.Http;
+using Raven.Client.Json.Serialization;
+using Sparrow.Json;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues;
+
+public class RavenDB_21490 : RavenTestBase
+{
+    public RavenDB_21490(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    [RavenFact(RavenTestCategory.Indexes | RavenTestCategory.Corax)]
+    private void OptimizeCoraxIndex()
+    {
+        using (var store = GetDocumentStore())
+        {
+            var index = new DummyCoraxIndex();
+                
+            index.Execute(store);
+                
+            Indexes.WaitForIndexing(store);
+                
+            var exception = Assert.Throws<NotSupportedInCoraxException>(() => store.Maintenance.Send(new IndexOptimizeOperation(index.IndexName)));
+
+            Assert.Contains("Index optimization is not supported in Corax.", exception.Message);
+        }
+    }
+    
+    [RavenFact(RavenTestCategory.Indexes | RavenTestCategory.Corax)]
+    private void DumpCoraxIndex()
+    {
+        var path = "dummy_path";
+        
+        using (var store = GetDocumentStore())
+        {
+            var index = new DummyCoraxIndex();
+                
+            index.Execute(store);
+                
+            Indexes.WaitForIndexing(store);
+
+            var exception = Assert.Throws<NotSupportedInCoraxException>(() => store.Maintenance.Send(new IndexDumpOperation(index.IndexName, path)));
+            
+            Assert.Contains("Index dump is not supported in Corax.", exception.Message);
+        }
+    }
+
+    private class Dto
+    {
+        public string Name { get; set; }
+    }
+    
+    private class DummyCoraxIndex : AbstractIndexCreationTask<Dto>
+    {
+        public DummyCoraxIndex()
+        {
+            Map = dtos => from dto in dtos
+                select new
+                {
+                    Name = dto.Name
+                };
+
+            SearchEngineType = Raven.Client.Documents.Indexes.SearchEngineType.Corax;
+        }
+    }
+    
+    private class IndexOptimizeOperation : IMaintenanceOperation<OperationIdResult>
```

Comment:
Instead of writing a dedicated command you'll be able to get index instance and call `Optimize()` and `Dump()` directly and they will throw

Reply:
Changed